client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/useInteractionsForm.ts

```diff
@@ -1,132 +1,181 @@
-            ...contact,
-            id: contact.idNumber
-          }))
-        })
-          .then(() => {
-            logger.info({
-              service: Service.CLIENT,
-              severity: Severity.LOW,
-              workflow: 'Update Interaction',
-              step: 'updated interaction successfully',
-              user: userId,
-              investigation: epidemiologyNumber
-            });
-            loadInteractions();
-            closeEditDialog();
-          }).catch((error) => {
-            logger.error({
-              service: Service.CLIENT,
-              severity: Severity.LOW,
-              workflow: 'Update Interaction',
-              step: `got error from server: ${error}`,
-              investigation: epidemiologyNumber,
-              user: userId
-            });
-            handleFailedSave('לא ניתן היה לשמור את השינויים');
-          })
-      } else {
-        logger.info({
-          service: Service.CLIENT,
-          severity: Severity.LOW,
-          workflow: 'Create Interaction',
-          step: `launching create interaction request`,
-          user: userId,
-          investigation: epidemiologyNumber
+    const checkForDuplicateIdsInContactEvent = (currContacts: Contact[]) => {
+        const allIdentificationNumbers = currContacts.map((contact) => contact.idNumber);
+        return allIdentificationNumbers.some((idNumber, idNumberIndex) => {
+            return allIdentificationNumbers.indexOf(idNumber) !== idNumberIndex
         });
-        axios.post('/intersections/createContactEvent', {
-          ...interactionsDataToSave,
-          [InteractionEventDialogFields.LOCATION_ADDRESS]: locationAddress,
-          [InteractionEventDialogFields.INVESTIGATION_ID]: epidemiologyNumber,
-          [InteractionEventDialogFields.CONTACTS]: interactionsDataToSave?.contacts?.map((contact: Contact) => ({
-            ...contact,
-            id: contact.idNumber
-          }))
-        })
-          .then(() => {
+    }
+
+    const saveInteractions = async (interactionsDataToSave: InteractionEventDialogData) => {
+        if(!checkForDuplicateIdsInContactEvent(interactionsDataToSave.contacts)) {
+            const locationAddress = interactionsDataToSave[InteractionEventDialogFields.LOCATION_ADDRESS] ?
+                await parseLocation(interactionsDataToSave[InteractionEventDialogFields.LOCATION_ADDRESS]) : null;
+            if (interactionsDataToSave[InteractionEventDialogFields.ID]) {
+                logger.info({
+                    service: Service.CLIENT,
+                    severity: Severity.LOW,
+                    workflow: 'Update Interaction',
+                    step: `launching update interaction request`,
+                    user: userId,
+                    investigation: epidemiologyNumber
+                });
+                axios.post('/intersections/updateContactEvent', {
```

Comment:
If you parse the data the same way on both update and create, maybe you should do it before the condition